diff --git a/library.js b/library.js
index 8ba5b21..20029e1 100644
--- a/library.js
+++ b/library.js
@@ -174,7 +174,8 @@ iframely.replace = function(raw, options, callback) {
 							title: embed.meta.title || url,
 							embed: embed,
 							icon: icon,
-							url: url
+							url: url,
+							date: validator.escape(embed.meta.date || '')
 						}, function (err, parsed) {
 							if (err) {
 								winston.error('[plugin/iframely] Could not parse embed: ' + err.message + '. Url: ' + url);
@@ -233,6 +234,7 @@ iframely.replace = function(raw, options, callback) {
 						favicon: wrapImage(icon),
 						embed: embed,
 						url: url,
+						date: validator.escape(embed.meta.date || ''),
 						metaString: metaInfo.length ? metaInfo.join('&nbsp;&nbsp;/&nbsp;&nbsp;') : false,
 						embedHtml: embedHtml,
 						embedIsImg: /^<img[^>]+>$/.test(embedHtml),
diff --git a/public/templates/partials/iframely-link-title.tpl b/public/templates/partials/iframely-link-title.tpl
index ee16400..95b9d74 100644
--- a/public/templates/partials/iframely-link-title.tpl
+++ b/public/templates/partials/iframely-link-title.tpl
@@ -1,6 +1,6 @@
 <div class="iframely-link">
     <div>
-        <a href="{url}" target="_blank" rel="nofollow noreferrer noopener">
+        <a href="{url}" target="_blank" rel="nofollow noreferrer noopener" data-date="{date}">
 
             <!-- IF icon -->
             <img src="{icon}" class="thumb pull-left not-responsive" />
diff --git a/public/templates/partials/iframely-widget-card.tpl b/public/templates/partials/iframely-widget-card.tpl
index b135b72..a065a32 100644
--- a/public/templates/partials/iframely-widget-card.tpl
+++ b/public/templates/partials/iframely-widget-card.tpl
@@ -10,7 +10,7 @@
 
             <!-- IF title -->
             <h4 class="media-heading">
-                <a href="{url}" target="_blank" rel="nofollow noreferrer noopener" class="one-line">
+                <a href="{url}" target="_blank" rel="nofollow noreferrer noopener" class="one-line" data-date="{date}">
                     <!-- IF favicon -->
                     <img src="{favicon}" class="thumb pull-left not-responsive" />
                     <!-- ENDIF favicon -->
