diff --git a/src/database/mongo.js b/src/database/mongo.js
index 5b264253f..952eb0cab 100644
--- a/src/database/mongo.js
+++ b/src/database/mongo.js
@@ -275,5 +275,6 @@ mongoModule.close = function (callback) {
 
 mongoModule.socketAdapter = function () {
 	var mongoAdapter = require('socket.io-adapter-mongo');
-	return mongoAdapter(mongoModule.getConnectionString());
+	var mubsub = require('mubsub-nbb');
+	return mongoAdapter(null, { client: mubsub(mongoModule.client) });
 };
diff --git a/src/database/mongo/sorted.js b/src/database/mongo/sorted.js
index d4eb7a852..59aa9f699 100644
--- a/src/database/mongo/sorted.js
+++ b/src/database/mongo/sorted.js
@@ -506,7 +506,8 @@ module.exports = function (db, module) {
 		var cursor = db.collection('objects').find({ _key: setKey })
 			.sort({ score: 1 })
 			.project(project)
-			.batchSize(options.batch);
+			.batchSize(options.batch)
+			.addCursorFlag('noCursorTimeout', true);
 
 		async.whilst(
 			function () {
