diff --git a/src/database/mongo.js b/src/database/mongo.js
index f16bd852a..f008aa832 100644
--- a/src/database/mongo.js
+++ b/src/database/mongo.js
@@ -91,6 +91,7 @@ function getConnectionString() {
 
 	return nconf.get('mongo:uri') || 'mongodb://' + usernamePassword + servers.join() + '/' + nconf.get('mongo:database');
 }
+mongoModule.getConnectionString = getConnectionString;
 
 mongoModule.init = function (callback) {
 	callback = callback || function () { };
@@ -272,5 +273,6 @@ mongoModule.close = function (callback) {
 
 mongoModule.socketAdapter = function () {
 	var mongoAdapter = require('socket.io-adapter-mongo');
-	return mongoAdapter(getConnectionString());
+	var mubsub = require('mubsub-nbb');
+	return mongoAdapter({ client: mubsub(getConnectionString()) });
 };
diff --git a/src/database/mongo/pubsub.js b/src/database/mongo/pubsub.js
index 83ff67629..cbb5b638b 100644
--- a/src/database/mongo/pubsub.js
+++ b/src/database/mongo/pubsub.js
@@ -1,8 +1,8 @@
 'use strict';
 
-var mubsub = require('mubsub');
+var mubsub = require('mubsub-nbb');
 
 var db = require('../mongo');
-var client = mubsub(db.client);
+var client = mubsub(db.getConnectionString());
 
 module.exports = client.channel('pubsub');
